mailchimp/mailchimp-authenticator/org.wso2.carbon.identity.authenticator/src/main/java/org/wso2/carbon/identity/authenticator/mailChimp/MailChimpAuthenticator.java

```diff
@@ -0,0 +1,306 @@
+            }
+            context.setProperty(OIDCAuthenticatorConstants.ACCESS_TOKEN, accessToken);
+            Map<ClaimMapping, String> claims = getSubjectAttributes(oAuthResponse, authenticatorProperties);
+            String email = claims.get(ClaimMapping.build(MailChimpAuthenticatorConstants.MailChimp_EMAIL, MailChimpAuthenticatorConstants.MailChimp_EMAIL, (String) null, false));
+            AuthenticatedUser authenticatedUserObj =
+                    AuthenticatedUser.createFederateAuthenticatedUserFromSubjectIdentifier(email);
+            authenticatedUserObj.setAuthenticatedSubjectIdentifier(email);
+            authenticatedUserObj.setUserAttributes(claims);
+            context.setSubject(authenticatedUserObj);
+        } catch (OAuthProblemException e) {
+            throw new AuthenticationFailedException("Authentication process failed", e);
+        }
+    }
+
+    @Override
+    protected Map<ClaimMapping, String> getSubjectAttributes(OAuthClientResponse token, Map<String, String> authenticatorProperties) {
+        HashMap claims = new HashMap();
+
+        try {
+            String e = token.getParam("access_token");
+            String url = this.getUserInfoEndpoint(token, authenticatorProperties);
+            String json = sendRequest(url, e);
+            JSONObject obj = new JSONObject(json);
+            if (StringUtils.isBlank(json)) {
+                if (log.isDebugEnabled()) {
+                    log.debug("Unable to fetch user claims. Proceeding without user claims");
+                }
+
+                return claims;
+            }
+
+            Map jsonObject = JSONUtils.parseJSON(json);
+            Iterator i$ = jsonObject.entrySet().iterator();
+
+            while (i$.hasNext()) {
+                Map.Entry data = (Map.Entry) i$.next();
+                String key = (String) data.getKey();
+                claims.put(ClaimMapping.build(key, key, (String) null, false), jsonObject.get(key).toString());
+                if (log.isDebugEnabled() && IdentityUtil.isTokenLoggable("UserClaims")) {
+                    log.debug("Adding claims from end-point data mapping : " + key + " - " + jsonObject.get(key).toString());
+                }
+            }
+        } catch (Exception var11) {
+            log.error("Error occurred while accessing user info endpoint", var11);
+        }
+
+        return claims;
+    }
+
+    protected String sendRequest(String url, String accessToken) throws IOException {
+        if (log.isDebugEnabled()) {
+            log.debug("Claim URL: " + url);
+        }
+
+        if (url == null) {
+            return "";
+        } else {
+            URL obj = new URL(url);
+            HttpURLConnection urlConnection = (HttpURLConnection) obj.openConnection();
+            urlConnection.setRequestMethod("POST");
+            HttpClient httpClient = HttpClientBuilder.create().build();
+            HttpPost p = new HttpPost(url);
```

Comment:
Fixed in https://github.com/wso2-attic/is-connectors/pull/69/commits/c9a7dd98c0c10e4efde3f0260cff94bb966477a4